packages/components/src/components/text/text.tsx

```diff
@@ -0,0 +1,48 @@
+import classNames from 'classnames';
+import React, { HTMLProps } from 'react';
+import { isEmptyObject } from '@deriv/shared';
+import { TGenericStringValueObject } from '../types';
+
+interface ITextProps extends HTMLProps<HTMLElement> {
+    align?: string;
+    weight?: string;
+    line_height?: string;
+    styles?: TGenericStringValueObject;
+}
+type TTextProps = Omit<ITextProps, 'size'> & {
+    size: string;
+};
+const Text = ({
+    children,
+    size = 's',
+    color = 'general',
+    align = 'left',
+    weight = 'normal',
+    line_height = 'm',
+    as,
+    className,
+    styles,
+    ...props
+}: TTextProps) => {
+    const class_styles = {
+        '--text-size': `var(--text-size-${size})`,
+        '--text-color': `var(--text-${color})`,
+        '--text-lh': `var(--text-lh-${line_height})`,
+        '--text-weight': `var(--text-weight-${weight})`,
+        '--text-align': `var(--text-align-${align})`,
+    };
+    const [style, setStyle] = React.useState(class_styles);
+    React.useEffect(() => {
```

Comment:
@jim-deriv Why do we need a state and effect for a simple text component? 🤔 
Whenever props change this component will rerender again with new props, Am I missing something? 🤔

Reply:
I agree with you @farzin-fs